### PR TITLE
review: fix: add handling for UnresolvedReferenceBinding in ReferenceBuilder

### LINKED ISSUE #5290: [Bug]:  Build model with source code from stdlib
### Describe the bug

I am trying to analyze the source code from the stdlib. Therefore I wrote an interface to retrieve the source code from the stdlib and add the content of the classes to the spoon launcher. It works for some classes e.g. `Object` but for some classes e.g. `Class` it seems to not work.

### Source code you are trying to analyze/transform

```Java
java.lang.Class
```

### Source code for your Spoon processing

```Java
Launcher launcher = new Launcher();
launcher.getEnvironment().setComplianceLevel(17);
launcher.getEnvironment().setIgnoreSyntaxErrors(true);

String className = "java.lang.Class";
String content = ...; // Source code of Class.java

VirtualFile file = new VirtualFile(content, className);


launcher.addInputResource(file);
CtModel model = launcher.buildModel();

CtElement element =  model.getAllTypes().stream()
                          .filter(it -> it.getQualifiedName().equals(className))
                          .findFirst()
                          .orElseThrow();
```

### Actual output

```Java
Unknown TypeBinding: class org.eclipse.jdt.internal.compiler.lookup.UnresolvedReferenceBinding Unresolved type java.lang.Class
java.lang.RuntimeException: Unknown TypeBinding: class org.eclipse.jdt.internal.compiler.lookup.UnresolvedReferenceBinding Unresolved type java.lang.Class
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:866)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:825)
	at spoon.support.compiler.jdt.ReferenceBuilder.getParameterizedTypeReference(ReferenceBuilder.java:900)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:840)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:825)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReferenceFromTypeArgument(ReferenceBuilder.java:938)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeArguments(ReferenceBuilder.java:923)
	at spoon.support.compiler.jdt.ReferenceBuilder.getParameterizedTypeReference(ReferenceBuilder.java:910)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:840)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:825)
	at spoon.support.compiler.jdt.ReferenceBuilder.getTypeReference(ReferenceBuilder.java:592)
	at spoon.support.compiler.jdt.ReferenceBuilder.buildTypeReference(ReferenceBuilder.java:138)
	at spoon.support.compiler.jdt.ReferenceBuilder.buildTypeReference(ReferenceBuilder.java:132)
	at spoon.support.compiler.jdt.JDTTreeBuilder.createParameterizedType(JDTTreeBuilder.java:1481)
	at spoon.support.compiler.jdt.JDTTreeBuilder.visit(JDTTreeBuilder.java:1469)
	at org.eclipse.jdt.internal.compiler.ast.ParameterizedSingleTypeReference.traverse(ParameterizedSingleTypeReference.java:405)
	at org.eclipse.jdt.internal.compiler.ast.Argument.traverse(Argument.java:275)
	at org.eclipse.jdt.internal.compiler.ast.MethodDeclaration.traverse(MethodDeclaration.java:427)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.traverse(TypeDeclaration.java:1831)
	at org.eclipse.jdt.internal.compiler.ast.TypeDeclaration.traverse(TypeDeclaration.java:1683)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.traverse(CompilationUnitDeclaration.java:827)
	at org.eclipse.jdt.internal.compiler.ast.CompilationUnitDeclaration.traverse(CompilationUnitDeclaration.java:788)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.traverseUnitDeclaration(JDTBasedSpoonCompiler.java:481)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.lambda$buildModel$0(JDTBasedSpoonCompiler.java:438)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.forEachCompilationUnit(JDTBasedSpoonCompiler.java:465)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.buildModel(JDTBasedSpoonCompiler.java:436)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.buildUnitsAndModel(JDTBasedSpoonCompiler.java:373)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.buildSources(JDTBasedSpoonCompiler.java:336)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:117)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:100)
	at spoon.Launcher.buildModel(Launcher.java:782)
```

### Expected output

```Java
No Error
```

### Spoon Version

10.3.0

### JVM Version

17.0.6

### What operating system are you using?

Windows 11

### PR DESCRIPTION
**Pull request by MartinWitt**
I am scared to add JDK code to the test resources. Any better suggestions? fixes #5290

### CODE REVIEW

**Comment by MartinWitt** [comment]
Given the current design, the new method has to return null. Any empty `CtTypeReference` is not helpful and hides the error.

File: src/test/resources/jdk/Class.java
Comment (I-Al-Istannen) [inline src/test/resources/jdk/Class.java:1]: I'd be for minimizing this and recreating the bug :^)

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]

**Comment by MartinWitt** [comment]
I haven't found any small reproducer, so lets close this and live with the bug.

**Comment by I-Al-Istannen** [comment]
I'd be in favor of deleting the test and merging it without any. The change is localized to an if that is not triggered by anything currently (or crashes), so we can't really make stuff worse.

**Review by I-Al-Istannen — APPROVED** [review]

**Pull request merged**